### PR TITLE
4 add logout

### LINKED ISSUE #4: add logout
- logout component on the frontend
- logout endpoint on the backend

### PR DESCRIPTION
**Pull request by erikgalajda**

### CODE REVIEW

**Comment by danielbubanec** [comment]
I think we'll need to merge this together because of my previous changes that are already in main branch 😂

File: apps/todoapp/src/app/todos/components/header/header.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <header class="header">
-  <todolist-user-info></todolist-user-info>
-  <!--Add Login/Logout button based on checkAuthStatus()-->
+  <todolist-user-info [user]="user"></todolist-user-info>
+  <app-logout *ngIf="user.username"></app-logout>
```
Comment (erikgalajda) [inline apps/todoapp/src/app/todos/components/header/header.component.html:3]: we'll have only logout here as login has it's own view

File: apps/todoapp/src/app/todos/components/todos/todos.component.ts [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
 
 @Component({
   selector: 'app-todos',
   templateUrl: './todos.component.html',
   styleUrls: ['./todos.component.scss'],
 })
-export class TodosComponent {}
\ No newline at end of file
+export class TodosComponent {
+  user!: DiscordUser;
```
Comment (erikgalajda) [inline apps/todoapp/src/app/todos/components/todos/todos.component.ts:11]: we load user here once and pass to all children that need it

File: apps/todoapp/src/app/todos/modules/main.module.ts [hunk 3]
```diff
@@ -24,6 +25,7 @@ const routes: Routes = [
     MainComponent,
     TodoComponent,
     FooterComponent,
+    LogoutComponent,
```
Comment (erikgalajda) [inline apps/todoapp/src/app/todos/modules/main.module.ts:28]: we should avoid renaming/restructuring/refactoring while implementing new features in parallel, it took me some time till I realize that this is missing here after it was renamed from todos -> main :)

**Comment by erikgalajda** [comment]
u can have a look, should be ready now, gonna sleep :)

**Pull request merged**